### PR TITLE
Explore: Fix browsing back to dashboard panel

### LINKED ISSUE #17030: Explore: browsing back to panel and forward to Explore broken
0. Dashboard with panel
1. Open the panel menu, click Explore
2. Wait until query loads in Explore, then hit browser Back button

Result: you stay on Explore. 
Expected: you get to the dashboard panel

Bonus: back/forward/back/forward/... should work as expected.

### PR DESCRIPTION
**Pull request by jschill**
**What this PR does / why we need it**:
Going to Explore from the panel menu redirects you to a new url with a new history entry instead of replacing existing one, so the back button wont take you back to the dashboard.

**Which issue(s) this PR fixes**:
Fixes #17030

**Special notes for your reviewer**:
Feels a little messy to pass that bool around. Maybe better to dispatch stateSave from initializeExplore instead?

### CODE REVIEW

**Comment by jschill** [comment]
@hugohaggmark Do you have a better way of solving this up your sleeve?

**Review by torkelo — APPROVED** [review]

**Comment by hugohaggmark** [comment]
I agree that it looks kind of messy, did you try to do this as you suggested in InitExplore? Although if this is holding up 6.2 release we could always refactor this later.

**Comment by jschill** [comment]
> I agree that it looks kind of messy, did you try to do this as you suggested in InitExplore? 

No, didn't try that. 

> Although if this is holding up 6.2 release we could always refactor this later.

👍 Good point

File: public/app/features/explore/state/actions.ts [hunk 1]
```diff
@@ -232,7 +231,7 @@ export function loadExploreDatasourcesAndSetDatasource(
     dispatch(loadExploreDatasources({ exploreId, exploreDatasources }));
 
     if (exploreDatasources.length >= 1) {
-      dispatch(changeDatasource(exploreId, datasourceName));
+      dispatch(changeDatasource(exploreId, datasourceName, true));
```
Comment (davkal) [inline public/app/features/explore/state/actions.ts]: I'd avoid using `true` as an argument and use a named variable instead.

**Review by davkal — COMMENTED** [review]
Works as well! Yeah not completely happy with the approach but I dont see a different one either. IMO can be merged as is.

**Pull request merged**